crypto/fipsmodule/service_indicator/service_indicator_test.cc

```diff
@@ -676,6 +671,317 @@ TEST_P(HMAC_ServiceIndicatorTest, HMACTest) {
   ASSERT_TRUE(check_test(hmacTestVector.expected_digest, mac.data(), mac_len, "HMAC KAT"));
 }
 
+TEST(ServiceIndicatorTest, RSAKeyGen) {
+ int approved = AWSLC_NOT_APPROVED;
+ bssl::UniquePtr<RSA> rsa(RSA_new());
+ ASSERT_TRUE(rsa);
+
+ // RSA_generate_key_fips may only be used for 2048-, 3072-, and 4096-bit
+ // keys.
+ ASSERT_FALSE(RSA_generate_key_fips(rsa.get(), 512, nullptr));
+ ASSERT_FALSE(RSA_generate_key_fips(rsa.get(), 1024, nullptr));
+ ASSERT_FALSE(RSA_generate_key_fips(rsa.get(), 3071, nullptr));
+ ASSERT_FALSE(RSA_generate_key_fips(rsa.get(), 4097, nullptr));
+
+ // Test that we can generate keys of the supported lengths:
+ for (const size_t bits : {2048, 3072, 4096}) {
+   SCOPED_TRACE(bits);
+
+   rsa.reset(RSA_new());
+   CALL_SERVICE_AND_CHECK_APPROVED(approved,
+               ASSERT_TRUE(RSA_generate_key_fips(rsa.get(), bits, nullptr)));
+   ASSERT_EQ(approved, AWSLC_APPROVED);
+   ASSERT_EQ(bits, BN_num_bits(rsa->n));
+ }
+}
+
+struct RSATestVector {
+  // key_size is the input rsa key size.
+  const int key_size;
+  // md_func is the digest to test.
+  const EVP_MD *(*func)(void);
+  // expected to be approved or not for signature generation.
+  const int sig_gen_expect_approved;
+  // expected to be approved or not for signature verification.
+  const int sig_ver_expect_approved;
+};
+struct RSATestVector kRSASigGenTestVectors[] = {
```

Comment:
For the name of the array, I think these vectors are for both Sign and Verify, but the name indicates `SigGen`